kratos/geometries/point_on_geometry.h

```diff
@@ -0,0 +1,378 @@
+        return Point(global_location);
+    }
+
+    /*
+    * @brief This method maps from dimension space to working space.
+    * @param rResult array_1d<double, 3> with the coordinates in working space
+    * @param LocalCoordinates The local coordinates in dimension space
+    * @return array_1d<double, 3> with the coordinates in working space
+    * @see PointLocalCoordinates
+    */
+    CoordinatesArrayType& GlobalCoordinates(
+        CoordinatesArrayType& rResult,
+        const CoordinatesArrayType& rLocalCoordinates
+    ) const override
+     {
+        KRATOS_ERROR << "Trying to access the GlobalCoordinates via PointOnGeometry. "
+            << "However, this is not possible. Try to access the global coordinates of"
+            << " any local coordinates via accessing the background geometry through: "
+            << "pGetGeometryPart(GeometryType::BACKGROUND_GEOMETRY_INDEX)." << std::endl;
+
+        return rResult;
+    }
+
+    ///@}
+    ///@name Integration Points
+    ///@{
+
+    /* Creates a integration point on the background geometry.
+     * @param return integration points list with one entry.
+     */
+    void CreateIntegrationPoints(
+        IntegrationPointsArrayType& rIntegrationPoints) const override
+    {
+        if (rIntegrationPoints.size() != 1) { rIntegrationPoints.resize(1); }
+        rIntegrationPoints[0][0] = mLocalCoordinates[0];
+        rIntegrationPoints[0][1] = mLocalCoordinates[1];
+        rIntegrationPoints[0][2] = mLocalCoordinates[2];
+        rIntegrationPoints[0].Weight() = 1.0;
+    }
+
+    ///@}
+    ///@name Quadrature Point Geometries
+    ///@{
+
+    /* @brief creates a list of quadrature point geometry
+     *        at the specific location of this PointOnGeometry on the background geometry.
+     *
+     * @param rResultGeometries list of ONE quadrature point geometries.
+     * @param rIntegrationPoints list of ONE integration point.
+     * @param NumberOfShapeFunctionDerivatives the number of evaluated
+     *        derivatives of shape functions at the quadrature point geometries.
+     *
+     * @see quadrature_point_geometry.h
+     */
+    void CreateQuadraturePointGeometries(
+        GeometriesArrayType& rResultGeometries,
+        IndexType NumberOfShapeFunctionDerivatives) override
+    {
+        IntegrationPointsArrayType integration_point(1);
+        this->CreateIntegrationPoints(integration_point);
+
+        GeometriesArrayType rQuadraturePointGeometries(1);
+        mpBackgroundGeometry->CreateQuadraturePointGeometries(
+            rQuadraturePointGeometries, NumberOfShapeFunctionDerivatives, integration_point);
+
+        if (rResultGeometries.size() != 1) { rResultGeometries.resize(1); }
+        // assignment operator for quadrature point geometry with Dimension being 0.
+        rResultGeometries[0] = QuadraturePointGeometry<
+            PointType, TWorkingSpaceDimension, TLocalSpaceDimension,
+            0>(rQuadraturePointGeometries[0]);
```

Comment:
@rickyaristio commenting those lines would make the code work as it is.

Reply:
Okay! I'll check this one.